cli/commands/helpers/yaml.go

```diff
@@ -0,0 +1,56 @@
+package helpers
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/sensu/sensu-go/types"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// wrapper is used to get the precise yaml encoding behaviour we want
+type wrapper struct {
+	Type string                 `yaml:"type"`
+	Spec map[string]interface{} `yaml:"spec"`
+}
+
+func wrapResource(r types.Resource) wrapper {
+	wrapped := types.WrapResource(r)
+	w := wrapper{
+		Type: wrapped.Type,
+		Spec: toMap(wrapped.Value),
+	}
+	return w
+}
+
+// toMap produces a map from a struct by serializing it to JSON and then
+// deserializing the JSON into a map. This is done to preserve business logic
+// expressed in customer marshalers, and JSON struct tag semantics.
+func toMap(v interface{}) map[string]interface{} {
+	b, _ := json.Marshal(v)
+	result := map[string]interface{}{}
+	_ = json.Unmarshal(b, &result)
+	return result
+}
+
+// PrintYAML serializes the value v to yaml and writes the result to w.
+func PrintYAML(v interface{}, w io.Writer) (err error) {
+	enc := yaml.NewEncoder(w)
+	defer func() {
+		if err == nil {
+			err = enc.Close()
```

Comment:
If `err != nil` you don't call `enc.Close()`.
Is that OK?

Reply:
Sure, I don't care what happens to the output stream if serialization has failed.